### PR TITLE
feat(grid): update how to use action-strip

### LINKED ISSUE #3007: Create Grid -> Row actions topic as part of the Grids section
This topic should focus on explaining how to use the Action strip in order to achieve a Gmail scenario with row actions for delete, edit and etc.

@hanastasov we can create a new demo for that topic as well.

The end goal is to showcase the actions strip usage in the context of the Grid, as we are exposing this feature in the App Builder as Row Actions.

### PR DESCRIPTION
**Pull request by Svetloslav15**
Closes #3007 

This PR needs the sample from [here](https://github.com/IgniteUI/igniteui-angular-samples/pull/2838), before merging.

### Checklist:
 
 - [ ] check topic's TOC/menu and paragraph headings
 - [ ] link to other topics using `../relative/path.md`
 - [ ] at the References section at the end of the topic add links to topics, samples, etc
 - [ ] reference API documentation instead of adding a section with API

<br />

 - [ ] use valid component names - [Data] Grid, `IgxSelectComponent`, `<igx-combo>`
 - [ ] use spell checker tool (VS Code, Grammarly, Microsoft Editor)
 - [ ] add inline `code blocks` for the names of classes / tags / properties
 - [ ] add language descriptor for the ```code blocks```
 - [ ] check broken links (use browser add-on)
 - [ ] check if sample is working and fully visible in the topic
 - [ ] check if sample is working and fully visible in the StackBlitz
 - [ ] check if code blocks match the code in StackBlitz demo

<br />

 - [ ] follow SEO guidelines to fill topic's metadata ([SEO guidelines](https://infragisticsinc297.sharepoint.com/:w:/g/Groups/marketing/EWz9InT4FDlErHCumxsKGY4Bd8H03yhRWxDFk47luRz-_Q?e=S5wWcx))

<br />

 - [ ] do not resolve requested changes (leave that to the reviewer)
 - [ ] add `pending-localization` label when the review of the PR is done
 - [ ] add a member from the localization team to translate it

### CODE REVIEW

File: en/components/action-strip.md [hunk 1]
```diff
@@ -94,12 +94,30 @@ This can be utilized via grid action components and we are providing two default
 
 >Note: When `IgxActionStripComponent` is a child component of the grid, hovering a row will automatically show the UI.
 
+### Row actions topic as part of the Grids section
```
Comment (hanastasov) [inline en/components/action-strip.md:97]: Not needed to take the title so literally ;) Also, action-strip.md topic is not part of the Grids section - look where the grid topics are located and create a new one, where to put this content.

**Review by hanastasov — CHANGES_REQUESTED** [review]

File: en/components/action-strip.md [hunk 2]
```diff
@@ -94,12 +94,30 @@ This can be utilized via grid action components and we are providing two default
...
+### Row actions topic as part of the Grids section 
+
+You can use the default ones or create custom component and then add it to the `ActionStrip`. If you want to use `ActionStrip` to achieve a Gmail scenario with row actions such as `delete`, `edit` and etc. You can simply create button component with `igx-icon`, add click event to it and insert it into the `igx-action-strip` component.
+
+```html
+ <igx-action-strip #actionstrip>
+    <igx-grid-pinning-actions></igx-grid-pinning-actions>
+        <button title="Edit" igxButton="icon" igxRipple 
+            (click)='startEdit(actionstrip.context)'>
+    <igx-icon>edit</igx-icon>
```
Comment (kdinev) [inline en/components/action-strip.md:106]: indentation

**Review by kdinev — CHANGES_REQUESTED** [review]

**Comment by Aleksandyr** [comment]
@Svetloslav15 @hanastasov IMO this topic has to be part of the existing one for the [actionSstrip](https://www.infragistics.com/products/ignite-ui-angular/angular/components/action-strip).
There is plenty of information which is not mentioned in the topic like the custom implementation and that the current implementation extends **igxGridActionBaseDirective** which might be misleading for the users.

File: en/components/grid/action-strip.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
+---
+title: Angular Grid ActionStrip | Ignite UI for Angular | infragistics 
+_description: Check how easy it is to use ActionStrip in Grid component to achieve row actions
+_keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI widgets, Angular, Native Angular Components Suite, Native Angular Controls, Native Angular Components Library, Angular ActionStrip components, Angular ActionStrip directives, Angular ActionStrip controls
+---
+
+# Grid ActionStrip
+
+The Grid component in Ignite UI for Angular provides the ability to use `ActionStrip` component to achieve row actions.
```
Comment (Aleksandyr) [inline en/components/grid/action-strip.md:9]:
```suggestion
The grid component in Ignite UI for Angular provides the ability to use `ActionStrip` and utilize CRUD for row/cell components and row pinning. 
```

File: en/components/grid/action-strip.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
+---
+title: Angular Grid ActionStrip | Ignite UI for Angular | infragistics 
+_description: Check how easy it is to use ActionStrip in Grid component to achieve row actions
+_keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI widgets, Angular, Native Angular Components Suite, Native Angular Controls, Native Angular Components Library, Angular ActionStrip components, Angular ActionStrip directives, Angular ActionStrip controls
+---
+
+# Grid ActionStrip
+
+The Grid component in Ignite UI for Angular provides the ability to use `ActionStrip` component to achieve row actions. 
+By default, we are providing two grid action components:
```
Comment (Aleksandyr) [inline en/components/grid/action-strip.md:10]:
```suggestion
Twо grid actions are provided by default:
```

File: en/components/grid/action-strip.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+- [`IgxGridEditingActionsComponent`]({environment:angularApiUrl}/classes/igxgrideditingactionscomponent.html) - includes functionality and UI related to grid editing. It allows you to quickly toggle edit mode for cells or rows, depending on the value of the `rowEditable` option of the grid and whether deleting rows is allowed.
```
Comment (Aleksandyr) [inline en/components/grid/action-strip.md:12]:
```suggestion
- [`IgxGridEditingActionsComponent`]({environment:angularApiUrl}/classes/igxgrideditingactionscomponent.html) - includes functionality and UI specifically designed for the grid editing. It allows you to quickly toggle edit mode for cells or rows, depending on the [`rowEditable`]({environment:angularApiUrl}/classes/IgxGridComponent.html#rowEditable) option and row deletion of the grid.
```

File: en/components/grid/action-strip.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+- [`IgxGridPinningActionsComponent`]({environment:angularApiUrl}/classes/igxgridpinningactionscomponent.html) - includes functionality and UI related to grid row pinning. It allows you to quickly pin rows and navigate between pinned rows and their disabled counterparts.
```
Comment (Aleksandyr) [inline en/components/grid/action-strip.md:13]:
```suggestion
- [`IgxGridPinningActionsComponent`]({environment:angularApiUrl}/classes/igxgridpinningactionscomponent.html) - includes functionality and UI specifically designed for the grid row pinning. It allows you to quickly pin rows and navigate between pinned rows and their disabled counterparts.
```

File: en/components/grid/action-strip.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    </igx-column>
+
+    <igx-action-strip #actionStrip>
+        <igx-grid-pinning-actions></igx-grid-pinning-actions>
+        <igx-grid-editing-actions></igx-grid-editing-actions>
+    </igx-action-strip>
+</igx-grid>
+```
+
+You can use the them or create custom component and then add it to the `ActionStrip`. If you want to use `ActionStrip` to achieve a Gmail scenario with row actions such as `delete`, `edit` and etc. You can simply create button component with `igx-icon`, add click event to it and insert it into the `igx-action-strip` component.
```
Comment (Aleksandyr) [inline en/components/grid/action-strip.md:27]:
```suggestion
These components expose templates giving flexibility for customization. For instance, if we would like to use the `ActionStrip` for a Gmail scenario with row actions such as `delete`, `edit` and etc. You can simply create button component with `igx-icon`, add click event to it and insert it into the `igx-action-strip` component.
```

**Review by Aleksandyr — CHANGES_REQUESTED** [review]

**Comment by hanastasov** [comment]
> @Svetloslav15 @hanastasov IMO this topic has to be part of the existing one for the [actionSstrip](https://www.infragistics.com/products/ignite-ui-angular/angular/components/action-strip). There is plenty of information that is not mentioned in the topic like the custom implementation and that the current implementation extends **igxGridActionBaseDirective** which might be misleading for the users.

It is not a problem to duplicate the content in the original topic as well, but the initial requirement that made us open this PR is the demand to have such a topic under Grid section.

Of course the content in the grid section will be probably less than the original one, as it only needs to cover what is relevant for action strip within the grid.

**Review by Aleksandyr — CHANGES_REQUESTED** [review]

**Review by Aleksandyr — APPROVED** [review]

File: en/components/grid/action-strip.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+ <igx-action-strip #actionstrip>
+    <igx-grid-pinning-actions></igx-grid-pinning-actions>
+        <button title="Edit" igxButton="icon" igxRipple (click)='startEdit(actionstrip.context)'>
+            <igx-icon>edit</igx-icon>
+        </button>
+        <button title="Delete" igxButton="icon" igxRipple *ngIf='!isDeleted(actionstrip.context)' (click)='actionstrip.context.delete()'>
+            <igx-icon>delete</igx-icon>
+        </button>
+    </igx-action-strip>
+</igx-grid>
```
Comment (kdinev) [inline en/components/grid/action-strip.md:44]:
```suggestion
<igx-grid>
    <igx-action-strip #actionstrip>
        <igx-grid-pinning-actions></igx-grid-pinning-actions>
        <button title="Edit" igxButton="icon" igxRipple (click)='startEdit(actionstrip.context)'>
            <igx-icon>edit</igx-icon>
        </button>
        <button title="Delete" igxButton="icon" igxRipple *ngIf='!isDeleted(actionstrip.context)' (click)='actionstrip.context.delete()'>
            <igx-icon>delete</igx-icon>
        </button>
    </igx-action-strip>
</igx-grid>
```

**Review by kdinev — CHANGES_REQUESTED** [review]

**Comment by kdinev** [comment]
Btw, I see that this is added to the `igx-grid` docs, but isn't the action strip usable within all our grids (flat, tree, hierarchical) and shouldn't this article be added to the grid template ones, hence be made available to all three grids?

**Review by hanastasov — CHANGES_REQUESTED** [review]
As already requested, content amy be moved under grids_templates, this way it will be reused among all grids.

Also, update the toc.yml files to include the new topic with NEW label.

PR should target vNext.

**Pull request closed without merging**